mem/basic.md

```diff
@@ -54,6 +54,14 @@ impl Memory for BasicMemory {
 }
 ```
 
+```rust
+impl Memory for BasicMemory {
+    fn new() -> Self {
+        Self { allocations: List::new() }
+    }
+}
+```
```

Comment:
Do you prefer `Memory: Default` and `#[derive(Default)]` to make this a bit more concise?

Reply:
I think I prefer `new`, so this looks good :)